pkg/operator/controllers/base/aro_controller.go

```diff
@@ -0,0 +1,105 @@
+package base
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+type AROController struct {
+	Log    *logrus.Entry
+	Client client.Client
+	Name   string
```

Comment:
Do these need to be public, or can we get by with defining a public constructor? [It looks like it should be](https://go.dev/play/p/PMd7FY-82ju?v=goprev), and that would reduce the diff on this PR (and all subsequent PRs) since this has downstream effects, namely on all access to `log` and `client` objects across every controller we manipulate with these changes.

I originally marked this as a nit, but after checking the first 5 controllers (all of them use `log` and `client`) I think this will save us a sufficient amount of work down the line to warrant changing here once now.

Reply:
My previous ("clunky") version of this, that I unfortunately force-pushed over, did use a public constructor, but this basically involved doubling up the fields on the base controller and component controllers. While this could be abstracted away behind the component controllers' public constructors, it was hard to follow.